sdk/core/azure_core_test/src/tracing.rs

```diff
@@ -174,34 +177,40 @@ impl AsAny for MockSpan {
     }
 }
 
+/// Expected information about a tracer.
 #[derive(Debug)]
 pub struct ExpectedTracerInformation<'a> {
+    /// Expected name for the tracer.
     pub name: &'a str,
+
+    /// Expected version for the tracer.
     pub version: Option<&'a str>,
+
+    /// Expected namespace for the tracer.
     pub namespace: Option<&'a str>,
-    pub spans: Vec<ExpectedSpanInformation<'a>>,
-}
 
-#[derive(Debug)]
-pub struct ExpectedSpanInformation<'a> {
-    pub span_name: &'a str,
-    pub status: SpanStatus,
-    pub kind: SpanKind,
-    pub attributes: Vec<(&'a str, AttributeValue)>,
+    /// A set of spans which should have been generated.
+    pub spans: Vec<ExpectedSpanInformation<'a>>,
 }
 
 pub fn check_instrumentation_result(
```

Comment:
Should be documented, unless it doens't need to be public. Is this also intended to be used in tests? I thought that was the `test_instrumentation_for_api` you expose.

Reply:
This is for a different set of tests, structured around the actual underlying OpenTelemetry API surface (the new API is structured around a service client API surface - the caller should have minimal knowledge of OpenTelemetry (that's the whole point of the new test API - it enables OpenTelemetry testing without actually knowing about the nitty gritty details).

But I'll document them.